qualtran/bloqs/generalized_qsp.py

```diff
@@ -79,6 +79,88 @@ def decompose_from_registers(
         yield cirq.GlobalPhaseGate(matrix[0, 0].conj()).on()
 
 
+def qsp_complementary_polynomial(
+    P: Sequence[complex], *, verify: bool = False
+) -> Sequence[complex]:
+    r"""Computes the Q polynomial given P
+
+    Computes polynomial $Q$ of degree at-most that of $P$, satisfying
+
+        $$ \abs{P(e^{i\theta})}^2 + \abs{Q(e^{i\theta})}^2 = 1 $$
+
+    for every $\theta \in \mathbb{R}$.
+
+    The exact method for computing $Q$ is described in the proof of Theorem 4.
```

Comment:
Maybe add a comment saying that this function computes the complementary polynomial using root finding following the ideas laid out in Theorem 4. But there's also a cheaper optimization based approach described in Eq 52

Reply:
added a todo note for eq. 52